CMakeLists.txt

```diff
@@ -5,62 +5,90 @@
 
 cmake_minimum_required(VERSION 3.20)
 cmake_policy(SET CMP0057 NEW)
-project(YOMM2 LANGUAGES CXX VERSION 1.5.2)
+
+project(YOMM2 LANGUAGES CXX VERSION 1.6.0)
 
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 17)
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-option(YOMM2_DOWNLOAD_DEPENDENCIES "Set to ON to build missing dependencies from source" OFF)
-include(cmake/download_package.cmake)
+option(YOMM2_SHARED "Build yomm2 as a shared library" OFF)
+option(YOMM2_ENABLE_EXAMPLES "Set to ON to build examples" OFF)
+option(YOMM2_DEBUG_MACROS "Set to ON to debug macros" OFF)
+option(YOMM2_ENABLE_DOC "Set to ON to generate tutorials and reference" OFF)
+option(YOMM2_ENABLE_TESTS "Set to ON to build tests" OFF)
+option(YOMM2_ENABLE_BENCHMARKS "Set to ON to enable benchmarks" OFF)
+option(
+  YOMM2_CHECK_ABI_COMPATIBILITY
+  "Build shared library and examples in different modes" OFF)
 
-# Find Boost dependency
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-find_package(Boost 1.74 QUIET)
-if(NOT TARGET Boost::headers)
-  if (YOMM2_DOWNLOAD_DEPENDENCIES)
-    download_package(Boost INSTALL_WITH_YOMM)
+include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(YOMM2_ENABLE_BENCHMARKS
+  "Set to ON to build benchmarks" OFF
+  "YOMM2_ENABLE_TESTS" OFF
+)
+
+macro(assign_bool VAR)
+  if(${ARGN})
+    set(${VAR} ON)
   else()
-    message(FATAL_ERROR  "Boost was not found on your system. Set YOMM2_DOWNLOAD_DEPENDENCIES to ON to download and build from the sources.")
+    set(${VAR} OFF)
   endif()
+endmacro()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(COMPILER_IS_CLANG ON)
 endif()
-message(STATUS "Using Boost libraries from ${Boost_INCLUDE_DIRS}")
 
-if(NOT ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug") AND (CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
-  add_compile_options(-save-temps -fverbose-asm -masm=intel)
+if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+  set(VARIANT_IS_DEBUG ON)
+endif()
+
+if (VARIANT_IS_DEBUG)
+  if(COMPILER_IS_CLANG)
+    add_compile_options(-fno-limit-debug-info)
+  endif()
+  if(CMAKE_COMPILER_IS_GNUCXX OR COMPILER_IS_CLANG)
+    add_compile_definitions(_GLIBCXX_DEBUG)
```

Comment:
This would add the definition on Windows if Clang is detected, not sure it matters though